### PR TITLE
UHF-79 Language switcher

### PR DESCRIPTION
**Pull request by khalima**
How to test (if you have no setup etc.):
- `composer create-project City-of-Helsinki/drupal-helfi-platform:dev-main hel-platform --no-interaction --repository https://repository.drupal.hel.ninja/`
- Switch to corresponding branches: 
   - `composer require drupal/hdbt:dev-UHF-79_language_switcher && composer require drupal/helfi_platform_config:dev-UHF-79_language_switcher`
- Run `make new`
- Go to https://hel-platform.docker.sh/fi/node/1
- Check that there is a language switcher / language switching links in the "universal header".
   - Check that the language switcher works in mobile and with tab & enter key

Also check these PRs:
- https://github.com/City-of-Helsinki/drupal-hdbt-admin/pull/11/files
- https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/23/files
- https://github.com/City-of-Helsinki/drupal-helfi/pull/34/files

### CODE REVIEW

**Review by teroelonen — CHANGES_REQUESTED** [review]
Otherwise works and looks good BUT on the mobile if I click it with mouse the dropdown opens and instantly closes again :(

File: src/scss/components/navigation/_language-switcher.scss [hunk 1]
```diff
@@ -0,0 +1,105 @@
+.language-switcher {
+  position: relative;
+
+  .language-link {
+    @include font-size(16px);
+    @include font-weight($font-weight-regular);
```
Comment (teroelonen) [inline src/scss/components/navigation/_language-switcher.scss:6]: Is this wrapper required? I think just the variable does the same thing. I don't remember if I have asked this before but I will ask it again :D

File: src/scss/components/navigation/_language-switcher.scss [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    color: $color-black;
+    text-decoration: none;
+
+    @include breakpoint($breakpoint-m) {
+      @include font-size(14px);
+      margin: 0 $spacing-half;
+    }
+
+    &.is-active {
+      @include font-weight($font-weight-bold);
```
Comment (teroelonen) [inline src/scss/components/navigation/_language-switcher.scss:16]: Same question here as above.

File: templates/navigation/links--language-block.html.twig [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+ *   removes it from screen readers and assistive technology. Headings allow
+ *   screen reader and keyboard only users to navigate to or skip the links.
+ *   See http://juicystudio.com/article/screen-readers-display-none.php and
+ *   http://www.w3.org/TR/WCAG-TECHS/H42.html for more information.
+ *
+ * @see template_preprocess_links()
+ */
+#}
+
+{% if links -%}
```
Comment (teroelonen) [inline templates/navigation/links--language-block.html.twig:35]: What is the -%} and is it supposed to be like that that there is no {%- on the first beginning?

File: templates/navigation/links--language-block.html.twig [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      <span class="icon--angle-down" aria-hidden="true">
+        {% include "@hdbt/misc/icon.twig" with {icon: 'angle-down', label: current_language } %}
+      </span>
+      {% if current_language %}
+        <span class="label" aria-label="{{ 'Current language is'|t }} {{ current_language|capitalize }}">{{ current_language|capitalize }}</span>
+      {% endif %}
+    </button>
+    <div aria-hidden="true" id="language-switcher-dropdown" class="language-switcher__dropdown">
+      <div class="language-links">
+        {%- for item in links -%}
```
Comment (teroelonen) [inline templates/navigation/links--language-block.html.twig:50]: Here is the mystical {%- -%} as well!!

File: templates/navigation/links--language-block.html.twig [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+          {% set language_link = '' %}
+          {% set lang = item.link['#options']['#abbreviation'] %}
+          {% set untranslated = item.link['#options']['#untranslated'] %}
+          {% set alternative_language = item.link['#options']['#alternative_language'] %}
+          {% set classes = ['language-link'] %}
+
+          {# Check if link is part of alternative menu and set class accordingly. #}
+          {% set classes = classes|merge([alternative_language ? 'is-alternative' : '']) %}
+
+          {%- if not untranslated and lang != current_langcode -%}
```
Comment (teroelonen) [inline templates/navigation/links--language-block.html.twig:60]: And here!

File: templates/navigation/links--language-block.html.twig [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            {% set language_link = path('<current>', {}, {'language': item.link['#options']['language']}) %}
+            {% set element = 'a' %}
+          {%- elseif lang == current_langcode -%}
```
Comment (teroelonen) [inline templates/navigation/links--language-block.html.twig:63]: And here!

File: templates/navigation/links--language-block.html.twig [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            {% set element = 'span' %}
+            {% set classes = classes|merge([lang == current_langcode ? 'is-active' : '']) %}
+          {%- else -%}
```
Comment (teroelonen) [inline templates/navigation/links--language-block.html.twig:66]: And here!

File: templates/navigation/links--language-block.html.twig [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+            {% set element = 'span' %}
+            {% set classes = classes|merge(['is-disabled']) %}
+          {%- endif -%}
```
Comment (teroelonen) [inline templates/navigation/links--language-block.html.twig:69]: And here!

File: templates/navigation/links--language-block.html.twig [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+          <{{ element }}
+            {{ create_attribute({'class': classes}) }}
+            {{ language_link ? create_attribute({'href': language_link}) }}
+            {{ element == 'a' ? create_attribute({'lang': lang}) }}
+          >{{ item.text|capitalize }}</{{ element }}>
+        {%- endfor -%}
+      </div>
+    </div>
+  </div>
+{%- endif %}
```
Comment (teroelonen) [inline templates/navigation/links--language-block.html.twig:82]: I am starting to see a pattern here! Now its only on the {%- and not on the last one?! What black magic is this?

**Review by teroelonen — CHANGES_REQUESTED** [review]
Added also some comments on the files :)

File: src/scss/components/navigation/_language-switcher.scss (hunk 1 above)
Comment (khalima) [reply src/scss/components/navigation/_language-switcher.scss:6]: Let's remove the mixin so we won't use it again.

File: templates/navigation/links--language-block.html.twig (hunk 3 above)
Comment (khalima) [reply templates/navigation/links--language-block.html.twig:35]: It's whitespace control, but that's obviously broken. 🤔

File: templates/navigation/links--language-block.html.twig (hunk 4 above)
Comment (khalima) [reply templates/navigation/links--language-block.html.twig:50]: Let's remove these and see if the whitespaces actually makes any difference.

**Review by teroelonen — APPROVED** [review]
This works correctly now :)

**Pull request merged**